substrate/frame/revive/src/exec.rs

```diff
@@ -1521,6 +1521,10 @@ where
 		storage_meter: &mut storage::meter::GenericMeter<T, S>,
 		exec_config: &ExecConfig<T>,
 	) -> DispatchResult {
+		if from == to || value.is_zero() {
```

Comment:
I assume this fix applies to pallet-revive as a whole and therefore it would fix this issue for EVM and PolkaVM. Is that correct?